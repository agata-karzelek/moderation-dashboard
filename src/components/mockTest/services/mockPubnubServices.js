import { mockAccounts } from "../mockAccounts";
import { mockPermissions } from "../mockPermissions";
import { mockPubNubApplications } from "../mockPubnubAccounts";

const commonkeyValuePairs = {
  status: 200,
  statusText: "OK",
  headers: {
    "cache-control": "no-cache, private",
    "content-type": "application/json",
    expires: "Thu, 29 Jul 2021 15:37:05 GMT",
    pragma: "private",
  },
  config: {
    url: "/apps?owner_id=572238",
    method: "get",
    headers: {
      Accept: "application/json, text/plain, */*",
      "X-Session-Token": "rAL18qRqj8rhtxkJwzQ2dmRi5UrhH-vFbqUaLknxshs",
    },
    baseURL: "https://admin.pubnub.com/api",
    transformRequest: [null],
    transformResponse: [null],
    timeout: 0,
    xsrfCookieName: "XSRF-TOKEN",
    xsrfHeaderName: "X-XSRF-TOKEN",
    maxContentLength: -1,
    maxBodyLength: -1,
  },
  request: {},
};

export const mockAllAccounts = {
  data: {
    result: {
      accounts: mockAccounts,
      permissions: mockPermissions,
    },
  },
  ...commonkeyValuePairs,
};

export const mockAllApps = {
  data: mockPubNubApplications,
  ...commonkeyValuePairs,
};

export default null;
