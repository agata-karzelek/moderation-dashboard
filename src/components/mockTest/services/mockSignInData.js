import { mockPermissions } from "../mockPermissions";

export const mockSignInData = {
  data: {
    result: {
      created: 1627276598,
      expires: 1627634943,
      modified: 1627548543,
      role: "user",
      status: 1,
      storage: null,
      token: "dGgB0-lLGG5GmjFgtr6t2nBFqgM05Nijq_YzzYdxVUk",
      user: {
        created: 1619760527,
        deleted: 0,
        email: "sourcefusepubnub@gmail.com",
        id: 572287,
        login_attempts: 0,
      },
      user_id: 572287,
      user_roles: {
        is_internal_admin: false,
        is_internal_operator: false,
        is_internal_viewer: false,
        roles: mockPermissions,
      },
    },
  },
  status: 200,
  statusText: "OK",
};

export default null;
