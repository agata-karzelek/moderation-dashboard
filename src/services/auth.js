import axios from "../utils/axios";

export async function signIn(credentials) {
  const signInResponse = await axios.post("/me", credentials);
  if (signInResponse.status === 200) {
    return signInResponse.data;
  }
  throw new Error(signInResponse.data.error || "Something went wrong");
}
