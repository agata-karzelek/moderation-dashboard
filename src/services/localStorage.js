import cookie from "js-cookie";

export const setCookie = (key, value) => {
  cookie.set(key, value, {
    expires: 1,
  });
};

export const removeCookie = (key) => {
  cookie.remove(key, {
    expires: 1,
  });
};

export const getCookie = (key) => {
  return cookie.get(key);
};

export const setLocalStorage = (key, value) => {
  localStorage.setItem(key, JSON.stringify(value));
};

export const removeLocalStorage = (key) => {
  localStorage.removeItem(key);
};

export const authenticate = (data, next) => {
  setCookie("token", data.token);
  setLocalStorage("user", data.user);
  next();
};

export const signout = (next) => {
  localStorage.clear();
  next();
};

export const isAuth = () => {
  const cookieChecked = getCookie("token");
  if (cookieChecked) {
    if (localStorage.getItem("user")) {
      return JSON.parse(localStorage.getItem("user"));
    } else {
      return false;
    }
  }
};
