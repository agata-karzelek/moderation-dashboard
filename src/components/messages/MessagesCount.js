import React, { useEffect, useState } from "react";
import { Typography } from "@material-ui/core";
import { getMessagesCount } from "../../services/pubnub";
import { useStyles } from "../../style/messages";
import SnackBar from "../core/SnackBar";
import { lastMidnightHours } from "../../utils/helpers";

export default function MessagesCount(props) {
  const classes = useStyles();
  const [count, setCount] = useState(0);
  const [alert, setAlert] = useState({
    success: { status: false, msg: "" },
    error: { status: false, msg: "" },
  });
  const midnightTimeToken = lastMidnightHours();

  useEffect(() => {
    setAlert({
      ...alert,
      success: { status: false, msg: "" },
      error: { status: false, msg: "" },
    });
    (async () => {
      try {
        const messagesCount = await getMessagesCount(
          props.pubnub,
          props.channelName,
          midnightTimeToken
        );
        setCount(messagesCount);
      } catch (e) {
        setAlert({
          ...alert,
          success: { status: false, msg: "" },
          error: { status: true, msg: e.message },
        });
      }
    })();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.channelName]);

  return (
    <>
      <Typography className={classes.messagesCountFont}>Today({count})</Typography>
      {alert.error.status && <SnackBar msg={alert.error.msg} status={"info"} />}
    </>
  );
}
