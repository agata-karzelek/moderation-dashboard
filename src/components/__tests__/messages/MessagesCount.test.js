import React from "react";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import MessagesCount from "../../messages/MessagesCount";
import { getMessagesCount } from "../../../services/pubnub";

jest.mock("../../../services/pubnub", () => ({
  getMessagesCount: jest.fn(),
}));

describe("Test Case For Messages Count", () => {
  test("Snapshot", async () => {
    getMessagesCount.mockImplementationOnce(() => 5);
    await act(async () => {
      const wrapper = mount(<MessagesCount channelName="pro.one" />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
