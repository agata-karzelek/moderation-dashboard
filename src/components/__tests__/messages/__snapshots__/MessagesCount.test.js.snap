// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case For Messages Count Snapshot 1`] = `
<MessagesCount
  channelName="pro.one"
>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-messagesCountFont-20"
  >
    <ForwardRef(Typography)
      className="makeStyles-messagesCountFont-20"
      classes={
        Object {
          "alignCenter": "MuiTypography-alignCenter",
          "alignJustify": "MuiTypography-alignJustify",
          "alignLeft": "MuiTypography-alignLeft",
          "alignRight": "MuiTypography-alignRight",
          "body1": "MuiTypography-body1",
          "body2": "MuiTypography-body2",
          "button": "MuiTypography-button",
          "caption": "MuiTypography-caption",
          "colorError": "MuiTypography-colorError",
          "colorInherit": "MuiTypography-colorInherit",
          "colorPrimary": "MuiTypography-colorPrimary",
          "colorSecondary": "MuiTypography-colorSecondary",
          "colorTextPrimary": "MuiTypography-colorTextPrimary",
          "colorTextSecondary": "MuiTypography-colorTextSecondary",
          "displayBlock": "MuiTypography-displayBlock",
          "displayInline": "MuiTypography-displayInline",
          "gutterBottom": "MuiTypography-gutterBottom",
          "h1": "MuiTypography-h1",
          "h2": "MuiTypography-h2",
          "h3": "MuiTypography-h3",
          "h4": "MuiTypography-h4",
          "h5": "MuiTypography-h5",
          "h6": "MuiTypography-h6",
          "noWrap": "MuiTypography-noWrap",
          "overline": "MuiTypography-overline",
          "paragraph": "MuiTypography-paragraph",
          "root": "MuiTypography-root",
          "srOnly": "MuiTypography-srOnly",
          "subtitle1": "MuiTypography-subtitle1",
          "subtitle2": "MuiTypography-subtitle2",
        }
      }
    >
      <p
        className="MuiTypography-root makeStyles-messagesCountFont-20 MuiTypography-body1"
      >
        Today(
        0
        )
      </p>
    </ForwardRef(Typography)>
  </WithStyles(ForwardRef(Typography))>
</MessagesCount>
`;
